lib/java-extras/src/test/java/org/triplea/io/ZipExtractorTest.java

```diff
@@ -24,43 +48,41 @@
    * Once the zip file is unzipped, the test verifies the files exist with expected contents.
    */
   @Test
-  void unzip() throws IOException {
-    final File destinationFolder = new File("destination");
-    destinationFolder.deleteOnExit();
-    try {
-      final File testDataZip = testDataZip();
-
-      ZipExtractor.unzipFile(testDataZip, destinationFolder);
-
-      assertThat(
-          "Should contain one folder and one file only",
-          FileUtils.listFiles(destinationFolder.toPath()),
-          hasSize(2));
-
-      final File unzippedTestFile1 = new File(destinationFolder, "test-file1");
-      final String testFile1Contents = Files.readString(unzippedTestFile1.toPath());
-      assertThat(testFile1Contents, is("ipsum ipsor\n"));
-
-      final File unzippedDirectory = new File(destinationFolder, "directory");
-      assertThat(unzippedDirectory.isDirectory(), is(true));
-      assertThat(unzippedDirectory.exists(), is(true));
-
-      final File unzippedTestFile2 = new File(unzippedDirectory, "test-file2");
-      final String testFile2Contents = Files.readString(unzippedTestFile2.toPath());
-      assertThat(testFile2Contents, is("lupsom osculus\n"));
-    } finally {
-      deleteDirectory(destinationFolder);
-    }
+  void unzip() throws IOException, URISyntaxException {
+    final Path testDataZip = testDataFolder.resolve("test-data.zip");
+
+    ZipExtractor.unzipFile(testDataZip, destinationFolder);
+
     assertThat(
-        "Ensure that we have no file handles open or anything else that"
-            + "would get in the way of deleting the unzipped folder",
-        destinationFolder.exists(),
-        is(false));
+        "Should contain one folder and one file only",
+        FileUtils.listFiles(destinationFolder),
+        hasSize(2));
+
+    final Path unzippedTestFile1 = destinationFolder.resolve("test-file1");
+    final String testFile1Contents = Files.readString(unzippedTestFile1);
+    assertThat(testFile1Contents, is("ipsum ipsor\n"));
+
+    final Path unzippedDirectory = destinationFolder.resolve("directory");
+    assertThat(Files.isDirectory(unzippedDirectory), is(true));
+    assertThat(Files.exists(unzippedDirectory), is(true));
+
+    final Path unzippedTestFile2 = unzippedDirectory.resolve("test-file2");
+    final String testFile2Contents = Files.readString(unzippedTestFile2);
+    assertThat(testFile2Contents, is("lupsom osculus\n"));
   }
 
-  private static File testDataZip() {
-    final File file =
-        new File(ZipExtractorTest.class.getClassLoader().getResource("test-data").getFile());
-    return new File(file, "test-data.zip");
+  @Test
```

Comment:
In order to add those tests I had to perform some slight refactoring to be able to test for the exception